D-Freezer/freezer.md

```diff
@@ -0,0 +1,42 @@
+# Implementing a scheduling class
+
+## Introduction
+One of the reasons students struggle with this assignment boils down to a lack of understanding of how the scheduler works. In this guide, I hope to provide you with a clear understanding of how the Linux scheduler pieces fit together. I hope to paint a picture that you can use to help you implement the freezer scheduler.
+
+## The `task_struct`
+
+Recall from HW1 that in Linux, every process is defined by its `struct task_struct`. When you have multiple tasks forked off a common parent, they are linked together in a doubly linked-list `struct list_head siblings` embedded within the `task_struct`. For example, if you had four processes running on your system, each forked off one parent, it will look something like this (the parent is not shown): 
+
+<div align='center'>
+    <img src='./task_struct.png'/><br/>
```

Comment:
nit: the middle blue arrow is missing a head

Reply:
forget this if youre running short on time

Reply:
not gonna have time to fix this